DFC.ServiceTaxonomy.GraphVisualiser/Controllers/VisualiseController.cs

```diff
@@ -91,18 +148,54 @@ private ActionResult GetOntology()
             return Content(owlResponseString, MediaTypeNames.Application.Json);
         }
 
-        private async Task<ActionResult> GetData(string uri, string graph)
+        private async Task<ActionResult> GetData(string contentItemId, string graph)
         {
-            const string prefLabel = "skos__prefLabel";
-            var query = new GetNodesCypherQuery(nameof(uri), uri, prefLabel, prefLabel);
+            IEnumerable<string> relationshipCommands = await BuildCommands(contentItemId);
 
-            //todo: allow user to visualise published and draft databases. new story?
-            await _neoGraphCluster.Run(graph, query);
+            var resultList = new List<INodeAndOutRelationshipsAndTheirInRelationships?>();
 
-            var owlDataModel = _neo4JToOwlGeneratorService.CreateOwlDataModels(query.SelectedNodeId, query.Nodes, query.Relationships, prefLabel);
-            var owlResponseString = JsonSerializer.Serialize(owlDataModel, _jsonOptions);
+            foreach (var command in relationshipCommands)
+            {
+                var result = await _neoGraphCluster.Run(graph, new NodeAndOutRelationshipsAndTheirInRelationshipsQueryFromCommand(command!));
```

Comment:
It'd be safer to run all the queries within a single transaction (there's a version of `GraphCluster.Run()` that accepts a set of queries).
That way the visualiser won't break if other users are altering the graph at the same time.

Reply:
The version that runs multiple accepts ICommands and has no return type. We could extend the method that accepts IQuery to take a collection?

Reply:
Ah yes.
If you wouldn't mind taking a look at creating an overload that accepts multiple queries, that'd be great.